pkg/kudoctl/cmd/repo_add.go

```diff
@@ -0,0 +1,112 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"strings"
+	"time"
+
+	"github.com/kudobuilder/kudo/pkg/kudoctl/kudohome"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/util/repo"
+
+	"github.com/gofrs/flock"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+type repoAddCmd struct {
+	name string
+	url  string
+	home kudohome.Home
+
+	out io.Writer
+	fs  afero.Fs
+}
+
+func (a repoAddCmd) run() error {
+	if err := addRepository(a.fs, a.name, a.url, a.home); err != nil {
+		return err
+	}
+	fmt.Fprintf(a.out, "%q has been added to your repositories\n", a.name)
+	return nil
+
+}
+
+func addRepository(fs afero.Fs, name string, url string, home kudohome.Home) error {
+	f, err := repo.LoadRepositories(fs, home.RepositoryFile())
+	if err != nil {
+		return err
+	}
+	if f.GetRepo(name) != nil {
+		return fmt.Errorf("repository name (%s) already exists, please specify a different name", name)
+	}
+	rc := &repo.RepositoryConfiguration{
+		URL:  url,
+		Name: name,
+	}
+	or, err := repo.NewOperatorRepository(rc)
+	if err != nil {
+		return err
+	}
+	_, err = or.DownloadIndexFile()
```

Comment:
It doesn't look like you're ever using the results of this function. Don't have it return an IndexFile if this is the case to clear up the function signature.

Reply:
this downloads the index file from the url of the repo.. and marshals into a `IndexFile`.   It confirms that the added repo is valid

Reply:
that's fine! does it need the return value for the IndexFile or do you just need an error here? no reason to return something that's not being used (and this func is only used here right now)